.github/workflows/publish.yml

```diff
@@ -1,59 +1,121 @@
-name: publish
+name: Version or Publish Package
 
 on:
-  push:
-    tags:
-      - 'v*.*.*'
-  workflow_dispatch: {}
+  workflow_dispatch:
+    inputs:
+      version_tag:
+        required: true
+        description: The release version
+        type: string
 
+concurrency: ${{ github.workflow }}-${{ github.ref }}
 
-# Declare default permissions as read only.
-permissions: read-all
+permissions:
+  id-token: write
+  contents: write
+  attestations: write
+  actions: write
+  pull-requests: write
```

Comment:
Top level let's keep read-all and for job specific we can give write permissions but we can do this later as an improvment